zcash_primitives/src/local_consensus.rs

```diff
@@ -0,0 +1,284 @@
+use crate::{
+    consensus::{BlockHeight, Network, NetworkUpgrade, Parameters},
+    constants,
+};
+
+/// Represents the Network Parameters of the Full Node the caller
+/// would be connecting to.
+/// `Consensus(Network::MainNetwork)` refers to a Zcash Full Node
+/// running on public Mainnet
+/// `Consensus(Network::TestNetwork)` refers to a Zcash Full Node
+/// running on public Testnet
+/// `Local(P)` can be used to define the consensus parameters of
+/// a local or non-public network like one or more nodes on
+/// `regtest` mode.
+pub enum FullNodeParameters<P: Parameters> {
+    Consensus(Network),
+    Local(P),
+}
+
+/// a `LocalNetwork` setup should define the activation heights
+/// of network upgrades. `None` is considered as "not activated"
+/// These heights are not validated. Callers shall initialized
+/// them according to the settings used on the Full Nodes they
+/// are connecting to.
+///
+/// Example:
+///     Regtest Zcashd using the following `zcash.conf`
+///     ```
+///     ## NUPARAMS
+///     nuparams=5ba81b19:1 # Overwinter
+///     nuparams=76b809bb:1 # Sapling
+///     nuparams=2bb40e60:1 # Blossom
+///     nuparams=f5b9230b:1 # Heartwood
+///     nuparams=e9ff75a6:1 # Canopy
+///     nuparams=c2d6d0b4:1 # NU5
+///     ```
+///     would use the following `LocalNetwork` struct
+///     ```
+///     let regtest = LocalNetwork {
+///         overwinter: Some(BlockHeight::from_u32(1)),
+///         sapling: Some(BlockHeight::from_u32(1)),
+///         blossom: Some(BlockHeight::from_u32(1)),
+///         heartwood: Some(BlockHeight::from_u32(1)),
+///         canopy: Some(BlockHeight::from_u32(1)),
+///         nu5: Some(BlockHeight::from_u32(1)),
+///         #[cfg(feature = "unstable-nu6")]
+///         nu6: Some(BlockHeight::from_u32(1)),
+///         #[cfg(feature = "zfuture")]
+///         z_future: Some(BlockHeight::from_u32(1)),
+///         };
+///     ```
+///     
+#[derive(Clone, PartialEq, Eq, Copy, Debug)]
+pub struct LocalNetwork {
+    pub overwinter: Option<BlockHeight>,
+    pub sapling: Option<BlockHeight>,
+    pub blossom: Option<BlockHeight>,
+    pub heartwood: Option<BlockHeight>,
+    pub canopy: Option<BlockHeight>,
+    pub nu5: Option<BlockHeight>,
+    pub nu6: Option<BlockHeight>,
+    #[cfg(feature = "zfuture")]
+    pub z_future: Option<BlockHeight>,
+}
+
+impl Parameters for LocalNetwork {
```

Comment:
Document that this `Parameters` impl hardcodes network constants other than activation heights to the `regtest` parameters.